### PR TITLE
[KMSv2] log request metadata as part of read/write

### LINKED ISSUE #111923: [KMSv2] Add additional logging with UID and metadata
Add additional logging at level `6` that associates UID with the metadata from the request.

xref: https://github.com/kubernetes/kubernetes/pull/111126#discussion_r934911719

/sig auth

### PR DESCRIPTION
**Pull request by aramase**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
- Logs additional metadata as part of transform `to_storage` and `from_storage`.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/111923

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
[KEP] https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3299-kms-v2-improvements
```

### CODE REVIEW

**Comment by enj** [comment]
/lgtm cancel

Looks like CI linting doesn't like building arguments.  Maybe create a separate function that returns a default request info if it's not present?  So the actual log statement would be able to use it unconditionally?

**Comment by aramase** [comment]
> /lgtm cancel
> 
> Looks like CI linting doesn't like building arguments. Maybe create a separate function that returns a default request info if it's not present? So the actual log statement would be able to use it unconditionally?

done!

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go [hunk 1]
```diff
@@ -109,8 +110,10 @@ func (t *envelopeTransformer) TransformFromStorage(ctx context.Context, data []b
 	if transformer == nil {
 		value.RecordCacheMiss()
 
+		requestInfo := getRequestInfoFromContext(ctx)
 		uid := string(uuid.NewUUID())
-		klog.V(6).InfoS("Decrypting content using envelope service", "uid", uid, "key", string(dataCtx.AuthenticatedData()))
+		klog.V(6).InfoS("decrypting content using envelope service", "uid", uid, "key", string(dataCtx.AuthenticatedData()),
+			"group", requestInfo.APIGroup, "version", requestInfo.APIVersion, "resource", requestInfo.Resource, "namespace", requestInfo.Namespace)
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go:116]: Sorry I missed this on the first pass - let's include the verb, sub resource, and name as well.

**Comment by sftim** [comment]
I think users could observe the outcome of this merge. Where'll we changelog it?

**Pull request merged**